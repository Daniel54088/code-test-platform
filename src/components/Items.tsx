import { useState } from 'react';
import { useDispatch, useSelector, RootStateOrAny } from 'react-redux';
import styled from 'styled-components';
import { ItemsProps,ItemBlockProps } from '../interfaces';
import moment from 'moment';
import NavigateNextIcon from '@mui/icons-material/NavigateNext';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';

const Items: React.FC<ItemsProps> = (props) => {

    const [clicked, handleClick] = useState(false);
    return (
        <Item>
            <ItemDetail>
                <ItemButtonArea>
                    {clicked ? (
                        <ButtonBorderClicked onClick={() => handleClick(false)}>
                            <KeyboardArrowDownIcon />
                        </ButtonBorderClicked>
                    ) : (

                        <ButtonBorder>
                            <NavigateNextIcon onClick={() => handleClick(true)} />
                        </ButtonBorder>
                    )}

                </ItemButtonArea>
                <ItemTitleArea>
                    <div>
                        <ItemTitleText>{props.data.title}</ItemTitleText>
                        <ItemDescriptionText>{props.data.id} | {props.data.description} </ItemDescriptionText>
                    </div>
                        <TabletLogo>  <img src={props.data.partner.logo} style={{ width: '100%'}} /> </TabletLogo>
                </ItemTitleArea>

                <ItemBlock >
                    <ItemBlockTitle>
                        {moment(props.data.payment_date).format('DD-MMM-YYYY')}
                    </ItemBlockTitle>
                    <ItemBlockDescription>
                        Payment date
                    </ItemBlockDescription>
                </ItemBlock>
                <Border />
                        
                {props.data.status === 'active' && (
                    <>
                        <ItemBlock mobileShows={true}>
                            <ItemBlockTitle>
                                {moment(Date.parse(props.data.coverage_start_date)).format('DD-MMM-YYYY')} to {moment(Date.parse(props.data.coverage_end_date)).format('DD-MMM-YYYY')}
                            </ItemBlockTitle>
                            <ItemBlockDescription>
                                Coverage dates
                            </ItemBlockDescription>
                        </ItemBlock>
                        <Border />
                    </>
                )}

                {props.data.status === 'expired' && (
                    <>
                        <ItemBlock mobileShows={true}>
                            <ItemBlockTitle>
                                {props.data.coverage_start_date}
                            </ItemBlockTitle>
                            <ItemBlockDescription>
                                Date shipped
                            </ItemBlockDescription>
                        </ItemBlock>
                        <Border />
                    </>
                )}

                <ItemBlock >
                    <ItemBlockTitle>
                        {props.data.premium_formatted}
                    </ItemBlockTitle>
                    <ItemBlockDescription>
                        Price/Premium
                    </ItemBlockDescription>
                </ItemBlock>


                {props.data.renewal !== null && (
                    <>
                        <Border />
                        <ItemBlock >
                            <ItemBlockTitle>
                                {props.data.coverage_start_date}
                            </ItemBlockTitle>
                            <ItemBlockDescription>
                                Renewal date
                            </ItemBlockDescription>
                        </ItemBlock>
                    </>
                )}

                <MobileLogo> 
                    <img src={props.data.partner.logo} style={{ width: '100%'}} /> 
                </MobileLogo>


            </ItemDetail>
            <FullLogo>
                <img src={props.data.partner.logo} />
            </FullLogo>
        </Item>
    );
}

export default Items;

const Item = styled.div`
    background: #FFFFFF;
    border-radius: 3px;
    display: flex;
    padding: 15px;
    margin-top: 18px;
    margin-bottom: 18px;
`;

const ItemDetail = styled.div`
    width: 80%;
    display: flex;
    flex-wrap: wrap;

    @media screen and (max-width: 820px) {
         width: 100%;    
    }
`;

const FullLogo = styled.div.attrs({ 'aria-label': 'full-logo' })`
    width: 20%;
    display: flex;
    justify-content: center;
    align-items: center; 

    @media screen and (max-width: 820px) {
        display: none;    
    }
`;

const ItemButtonArea = styled.div`
    width: 10%;
    border-bottom: 2px solid #E0E4E8;
    display: flex;
    justify-content: center;
    align-items: center; 
    padding-bottom: 12px;

    @media screen and (max-width: 480px) {
        display: none;
        
    }   
`;

const ButtonBorder = styled.div.attrs({ 'aria-label': 'button' })`
    width: 48px;
    height: 48px;
    border: 2px solid #FFD200;
    display: flex;
    justify-content: center;
    align-items: center; 
    border-radius: 50%;
    cursor:pointer;

    @media screen and (max-width: 480px) {
        display: none;
    }

`;

const ButtonBorderClicked = styled.div.attrs({ 'aria-label': 'clicked-button' })`
    width: 48px;
    height: 48px;
    border: 2px solid #FFD200;
    display: flex;
    background-color:#FFE600;
    justify-content: center;
    align-items: center; 
    border-radius: 50%;
    cursor:pointer;

    @media screen and (max-width: 480px) {
        display: none;
    }
`;


const ItemTitleArea = styled.div.attrs({ 'aria-label': 'title-area' })`
    width: 90%;
    border-bottom: 2px solid #E0E4E8;
    padding-bottom: 20px;
    display:flex;

    @media screen and (max-width: 480px) {
        width: 100%;
    }

    @media screen and (max-width: 820px) {
       padding-bottom: 6px;

    }
`;


const ItemTitleText = styled.div`
    font-family: Brandon Grotesque;
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    line-height: 30px;
    color: #2D2D2D;

    @media screen and (max-width: 480px) {
        font-size: 16px;
    }

    @media screen and (min-width: 481px) and (max-width: 820px) {
        font-size: 18px;
    }
`;

const ItemDescriptionText = styled.div`
    font-family: Bitter;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 21px;
    color: #73777C;

    @media screen and (max-width: 820px) {
        font-size: 12px;   
    }
`;

const ItemBlock = styled.div.attrs<ItemBlockProps>(
    ({  }) => ({
        'aria-label': 'item-block'
    })
  )<ItemBlockProps>`
    padding: 15px;
    &:before {
        content : "";
        position: absolute;
        left    : 0;
        bottom  : 0;
        height  : 1px;
        width   : 50%; 
        border-right:1px solid #E0E4E8;
      }

      @media screen and (max-width: 480px) {
        display: ${props => props.mobileShows ? "block" : "none"};
        padding: 10px 0px 10px 0px;
    }

`;

const ItemBlockTitle = styled.div`
    font-family: Brandon Grotesque;
    font-style: normal;
    font-weight: 390;
    font-size: 16px;
    line-height: 24px;
    color: #2D2D2D;
`;

const ItemBlockDescription = styled.div`
    font-family: Bitter;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 18px;
    color: #73777C;
`;

const Border = styled.div`
    border: 0.5px solid #E0E4E8;
    height: 40%;
    margin-top: 12px;

    @media screen and (max-width: 480px) {
        display: none;
    }
`

const TabletLogo = styled.div`
    display: none;

    @media screen and (max-width: 480px) {
        display: none;
    }
    
    @media screen and  (min-width: 481px) and (max-width: 820px) {
        display: flex;
        justify-content: center;
        align-items: center; 
        width: 20%;
        margin-left: auto;
    }

`

const MobileLogo = styled.div`
    display: none;
    @media screen and (max-width: 480px) {
        display: flex;
        width: 35%;
        margin-left: auto;
    }
`






