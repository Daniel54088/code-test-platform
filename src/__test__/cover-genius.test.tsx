import React from 'react';
import {render, fireEvent,  screen, RenderResult} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import Items from '../components/Items';

const correctDataFomat =
    {
        "id": "TRAVEL-COVER-INS",
        "type": "travel",
        "title": "London to Paris",
        "description": "Baggage Cover; Medical Cover; Cancellation Cover",
        "status": "active",
        "premium": 106.65,
        "premium_formatted": "AUD $106.65",
        "payment_date": "2019-10-10T13:29:38.814849Z",
        "coverage_start_date": "2019-11-17",
        "coverage_end_date": "2019-11-19",
        "renewal": null,
        "partner": {
          "id": "labore",
          "name": "Labore Inc.",
          "logo": "https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
        }
      };

let documentBody: RenderResult;

describe('<Items />', () => {

  beforeEach(() => {
    documentBody = render(<Items data={correctDataFomat} />);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });

  it('Title area texts are showing correctly', () => {
    const titleArea = screen.getByLabelText('title-area');
    expect(titleArea).toHaveTextContent('London to Paris');
    expect(titleArea).toHaveTextContent('Baggage Cover; Medical Cover; Cancellation Cover');
    expect(titleArea).toHaveTextContent('TRAVEL-COVER-INS');
   });

   it('Dates and price detail texts are showing correctly', () => {
    const itemBlocks = screen.getAllByLabelText('item-block');

    expect(itemBlocks.length).toBe(3);

    expect(itemBlocks[0]).toHaveTextContent('10-Oct-2019');
    expect(itemBlocks[1]).toHaveTextContent('17-Nov-2019 to 19-Nov-2019');
    expect(itemBlocks[2]).toHaveTextContent('AUD $106.65Price/Premium');
   });

   it('Logo is showing correctly', () => {
    const logo = screen.getByLabelText('full-logo');
    expect(logo).toHaveStyle(` display: flex`);
   });

   

});