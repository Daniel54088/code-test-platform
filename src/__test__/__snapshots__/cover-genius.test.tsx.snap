// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdvvtL dDAyCc"
    >
      <div
        class="sc-gsDKAQ dZzKlp"
      >
        <div
          class="sc-hKwDye dRHQYv"
        >
          <div
            aria-label="button"
            class="sc-eCImPb jnqXPd"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          aria-label="title-area"
          class="sc-gKclnd fvDSOG"
        >
          <div>
            <div
              class="sc-iCfMLu dNNagR"
            >
              London to Paris
            </div>
            <div
              class="sc-furwcr fmmRao"
            >
              TRAVEL-COVER-INS
               | 
              Baggage Cover; Medical Cover; Cancellation Cover
               
            </div>
          </div>
          <div
            class="sc-crHmcD gxgNqW"
          >
              
            <img
              src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
              style="width: 100%;"
            />
             
          </div>
        </div>
        <div
          aria-label="item-block"
          class="sc-pVTFL evGKuA"
        >
          <div
            class="sc-jrQzAO enDOMz"
          >
            10-Oct-2019
          </div>
          <div
            class="sc-kDTinF cCJqRp"
          >
            Payment date
          </div>
        </div>
        <div
          class="sc-iqseJM eZXJFy"
        />
        <div
          aria-label="item-block"
          class="sc-pVTFL ecqzjf"
        >
          <div
            class="sc-jrQzAO enDOMz"
          >
            17-Nov-2019
             to 
            19-Nov-2019
          </div>
          <div
            class="sc-kDTinF cCJqRp"
          >
            Coverage dates
          </div>
        </div>
        <div
          class="sc-iqseJM eZXJFy"
        />
        <div
          aria-label="item-block"
          class="sc-pVTFL evGKuA"
        >
          <div
            class="sc-jrQzAO enDOMz"
          >
            AUD $106.65
          </div>
          <div
            class="sc-kDTinF cCJqRp"
          >
            Price/Premium
          </div>
        </div>
        <div
          class="sc-egiyK gDumpy"
        >
          <img
            src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
            style="width: 100%;"
          />
        </div>
      </div>
      <div
        aria-label="full-logo"
        class="sc-dkPtRN ipSnZw"
      >
        
        <img
          src="https://s3-ap-southeast-2.amazonaws.com/cg-frontend-tests/labore.svg"
        />
      </div>
    </div>
  </div>
</body>
`;
